### DISCUSSION TITLE
Code autogeneration

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [space-ros/space-ros#146] [Q&A] [answered]
Are we using any tools to generate any of the code included in Space ROS? Are we aware of the ROS 2 team doing it, or any of the Space ROS dependencies doing it?

**Comment by Bckempa** [comment, marked as answer]
In general, (almost?) all ROS packages will use the [`rosidl`](https://github.com/ros2/rosidl) API which does code generation based on the `.msg` and `.srv` files during the build process. More docs [here](https://docs.ros.org/en/rolling/Concepts/Advanced/About-Internal-Interfaces.html#the-rosidl-repository).

**Reply by ivanperez-keera** [reply]
Thanks, Brian. That's exactly what I was looking for.
